pkg/evaluator/value.go

```diff
@@ -300,6 +323,27 @@ func (m *mapVal) String() string {
 	return "{" + strings.Join(pairs, " ") + "}"
 }
 
+func (m *mapVal) Repr() string {
+	pairs := make([]string, 0, len(m.Pairs))
+	for _, key := range *m.Order {
+		pairs = append(pairs, keyRepr(key)+":"+m.Pairs[key].Repr())
+	}
+	return "{" + strings.Join(pairs, " ") + "}"
+}
+
+func keyRepr(s string) string {
```

Comment:
food for thought. It seems to me perhaps there could be a function `lexer.IsIdent(s)` that this function could be written in terms of:
```
func keyRepr(s string) string {
  if lexer.IsIdent(s) {
    return s
  }
  return strconv.Quote(s)
}
```

Then the `lexer` package could have `IsIdent()` which makes sense as a place for this to belong (and it already has `isLetter(r)` (which allows `_`) and `isDigit(r)`:
```
func IsIdent(s) bool {
  if s == "" {
    return false
  }
  for i, r := range s {
    if !isLetter(r) && (i > 0 && !isDigit(r)) {
      return false
    }
  }
  return true
}
```

Reply:
I like this a lot! Adopted - thank you.